.github/workflows/apiclient-test.yaml

```diff
@@ -0,0 +1,36 @@
+name: Apiclient Test
+
+on:
+  push:
+    branches:
+      - master
+      - release-*
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      max-parallel: 2
```

Comment:
Why?

Reply:
Java is a beast and may need quite some resources, in my earlier testing in #163 I noticed that it sometimes went out of memory. Limiting the maximum amount of builds to 2 fixed it.

Reply:
That's strange. According to their [docs](https://docs.github.com/es/actions/reference/workflow-syntax-for-github-actions#) each run in a matrix runs inside their own worker.

[Workers have 7 GB as well](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) so I wonder how you could run out of memory as well

If you take a look at my [Jellyfin](https://github.com/ferferga/jellyfin) repo, where I'm working on the GH Actions migration for server for a future PR, you can see that I have a job that has a matrix for all the OSes and architectures, which dispatches 15 workers for a single job. [Maximum concurrent jobs are 20](https://docs.github.com/es/actions/reference/usage-limits-billing-and-administration), so that falls near the limits. 

I guess that max-parallel option was made for these kinds of scenarios like mine in server: If I push too quick in a row, the dispatched jobs only start after others have finished first. Some people might want their workflows to start as soon as possible, instead of parallelization, so that settings is tremendously useful for them.

However, imo, your setup isn't as "intensive" as the one I'm replacing in server or web ones (it's just a build process, no additional jobs for linting/testing like we have in Vue) so I don't see real benefit in removing parallelism, specially if there's no real proof that what you experienced before was a one-time thing or it could be reproduced consistently with the full parallel workflow.

Reply:
It might have changes, I'm quite sure they did run parallel on the same machine, sharing resources, in the past. Thanks for your explanation though. I'll remove this line :)